src/daft-micropartition/src/lib.rs

```diff
@@ -59,3 +63,8 @@ impl From<Error> for pyo3::PyErr {
         daft_error.into()
     }
 }
+
+pub trait FileWriter: Send + Sync {
+    fn write(&self, data: &Arc<MicroPartition>) -> DaftResult<()>;
+    fn close(&self) -> DaftResult<Option<String>>;
+}
```

Comment:
Idk where to put this trait, ideally somewhere that already has Daft-Micropartition as a dependency, so not Daft-IO.